src/tools/fetchDocPage/utils.ts

```diff
@@ -0,0 +1,112 @@
+import { FETCH_TIMEOUT_MS } from '../../constants.ts';
+import { logger } from '../../logger.ts';
+
+const getTimeout = (abortController: AbortController) =>
+  setTimeout(() => {
+    /* v8 ignore next */
+    abortController.abort();
+  }, FETCH_TIMEOUT_MS);
+
+/**
+ * Get the content of a remote documentation page.
+ *
+ * @param url - The URL of the remote documentation page to fetch.
+ */
+const getRemotePage = async (
+  url: URL
+): Promise<{ markdown: string; eTag: string | null }> => {
+  const abortController = new AbortController();
+  const timeoutId = getTimeout(abortController);
+  try {
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        Accept: 'text/markdown',
+      },
+      signal: abortController.signal,
```

Comment:
If you use the `AbortSignal` global the you don't have to do any of the `setTimeout` stuff or manually create the abort controller:

```ts
  // ...
  signal: AbortSignal.timeout(5000)
  // ...
```

Reply:
Sweet, I didn't know! Changing it.